### DISCUSSION TITLE
Understand Morgan fingerprint bit radius

### DISCUSSION THREAD

**Discussion by LiuCMU** [rdkit/rdkit#5826] [Q&A] [answered]
Hi, set the `radius=2` and get Morgan fingerprint for the following molecules. I visualized the bits, finding that some bits don't consider neighbors around 2 (for example, bits at 33 and 356). Does anyone know why? Thank you very much!
```
import numpy as np
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import Draw

smi = "CC1=CC=CC=C1"
mol = Chem.MolFromSmiles(smi)
```
![image](https://user-images.githubusercontent.com/60156077/205109591-949f35b5-4b32-4e96-bff1-4c8347ec1adc.png)

```
bi = {}
fp = np.array(AllChem.GetMorganFingerprintAsBitVect(mol, radius=2, bitInfo=bi, nBits=1024))
Draw.DrawMorganBits([(mol, 31, bi), (mol, 33, bi), (mol, 64, bi),
(mol, 175, bi), (mol, 356, bi), (mol, 389, bi),
(mol, 698, bi), (mol, 726, bi), (mol, 799, bi), 
(mol, 849, bi), (mol, 896, bi)], useSVG=True)
```
<img width="371" alt="image" src="https://user-images.githubusercontent.com/60156077/205109897-908b2402-d1dc-44d3-9431-bdc74999a3ad.png">

**Comment by manajit-das** [comment, marked as answer]
Hi LiuCMU, 
When you set the radius=2, it will give all the fragments for the radius 0 and radius 1 as well. Thus, radius=2, has the information of the fragments at radius0 and 1. Here are the pics showing the fragments at radius 0 and 1
![radius1](https://user-images.githubusercontent.com/59758360/205922229-5c3559fd-38d0-4829-932b-39e0cc81cc58.PNG)
![radius0](https://user-images.githubusercontent.com/59758360/205922235-a76a5da8-cd0f-4301-a310-6c340674649c.PNG)

**Reply by LiuCMU** [reply]
I see, this makes a lot of sense. Thank you!

BTW, I see that you used `rdMolDescriptors.GetMorganFingerprintAsBitVect`. I usually use `AllChem.GetMorganFingerprintAsBitVect`. Are they the same?

**Reply by manajit-das** [reply]
I have not checked if both are same or not!, but it seems same as both of them produced same result in terms of on and off bits.

**Reply by LiuCMU** [reply]
Got it, thank you!

**Reply by greglandrum** [reply]
> 
> BTW, I see that you used `rdMolDescriptors.GetMorganFingerprintAsBitVect`. I usually use `AllChem.GetMorganFingerprintAsBitVect`. Are they the same?

They are exactly the same. `AllChem` is a convenience module which imports a lot of the other RDKit modules.
You can check this yourself from inside python:
```
In [27]: AllChem.GetMorganFingerprintAsBitVect
Out[27]: <function rdkit.Chem.rdMolDescriptors.GetMorganFingerprintAsBitVect>
```

**Reply by LiuCMU** [reply]
I see, `AllChem` is a handy module. Thanks for the excellent work!
